certbot/reverter.py

```diff
@@ -587,3 +545,35 @@ def _timestamp_progress_dir(self):
             self.config.in_progress_dir, final_dir)
         raise errors.ReverterError(
             "Unable to finalize checkpoint renaming")
+
+
+def _read_json_file(filepath, formatting='plaintext'):
+    try:
+        # It is strongly advised to set newline = '' on Python 3 with CSV,
+        # and it fixes problems on Windows.
+        kwargs = {'newline': ''} if sys.version_info[0] > 2 and formatting == 'csv' else {}
+        with open(filepath, 'r', **kwargs) as file_h:  # type: ignore  # pylint: disable=star-args
+            if formatting == 'csv':
+                return list(csv.reader(file_h))
+            else:
+                # strip() will suppress the newlines characters in respect of the platform.
```

Comment:
I don't understand the need for this. See the documentation for the newline parameter of the open function at https://docs.python.org/3/library/functions.html#open. Isn't Python translating everything for us automatically?

Reply:
You are not looking to the right problem here. It is true that by default, `open` will convert from and to a file any platform specific end of lines characters. So you will, from the python code, always handle with `\n`. But here we need to extract, as a list, every line. So `newlines` is involved, and it is its behavior that is problematic here: it will retrieve each line, with a end of line character at the end. It may or may not be platform specific depending on how `open` is invoked, but it will keep being a problem because we need to remove that end of line character.

So we need to remove it, and `strip()` will do that, for any end of line characters (here always `\n` because we are in test/universal newline mode).

As an alternative of that, I could use the shorter `lines = file_h.read().splitlines()`, that will do the two jobs (iterate and strip) in one action.

Reply:
I use `splitlines`. It is the most simple and compatible way to extract lines from a file.